app/views/handlers/individual_response.py

```diff
@@ -139,7 +140,15 @@ def handle_get(self):
             self.individual_section_id
         )
 
-        if self._list_item_id:
+        if self._request_args.get("journey") == "remove-person":
+            previous_location_url = url_for(
+                "questionnaire.block",
+                list_name=self._list_name,
+                list_item_id=self._list_item_id,
+                block_id="remove-person",
+            )
```

Comment:
The journey needs to persist, at the moment, the previous works correct for only the first screen. If you progress forward, then essentially clicking previous multiple times won't take you back to where you came from hence when you proceed forward in the journey, it also needs to persist the journey param.